channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -283,6 +283,10 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
                     _syncUnloadSender = _fallbackSender;
                 }
             };
+
+            _self.setBuffer = (buffer: new (logger: IDiagnosticLogger, config: ISenderConfig) => ISendBuffer) => {
```

Comment:
Yeh, we don't (currently) really want to expose this as a public API.

Longer term (hopefully by the end of '23) we have a different approach that we want to take in relation to the buffers.
ie. As part of the internal code that we will be bringing over we abstract the storage out completely so it's not even included in the Sender but rather managed by a pluggable component and using the notifications...

Currently the Sender doesn't send notifications for all types -- so there is an amount of refactoring before we can bring that over.

So as part of this we probably want to consider am internal config type approach.

We had a team meeting about this one as well and we don't want to include this PR into the next release (2.8.12) as we are trying to released that late today / tomorrow. But we are silling for 2.8.13 (which would be next month)

Reply:
The rationale about delaying is that we are also trying to get v3.0.0 out as well and this would require quite a bit of churn for the first release.

In terms of a config approach defining as a `bufferOverride` with a get/set would probably work nicely now and as part of future refactoring...

Reply:
I've removed this and gone with your suggestion.

To enable local storage, you now must set `enableSessionStorageBuffer` to `true` AND then specify the `bufferOverride`.

`bufferOverride` is just a simple string for now, but should be easy to extend for the future.

Example usage might be:
```
const appInsights = new ApplicationInsights({
  config: {
    enableSessionStorageBuffer: true,
    bufferOverride: "localStorage"
  }
});
```